GitUI/UserControls/InteractiveGitActionControl.cs

```diff
@@ -0,0 +1,240 @@
+            {
+                case State.Rebase:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ContinueRebaseCmd());
+                    break;
+                case State.Merge:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ContinueMergeCmd());
+                    break;
+                case State.Patch:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.ResolvedCmd());
+                    break;
+                default: return;
+            }
+
+            form.UICommands.RepoChangedNotifier.Notify();
+        }
+
+        private void AbortButton_Click(object sender, EventArgs e)
+        {
+            if (!(FindForm() is GitModuleForm form))
+            {
+                return;
+            }
+
+            switch (_state)
+            {
+                case State.Rebase:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.AbortRebaseCmd());
+                    break;
+                case State.Merge:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.AbortMergeCmd());
+                    break;
+                case State.Patch:
+                    FormProcess.ShowDialog(form, GitCommandHelpers.AbortCmd());
+                    break;
+                default: return;
+            }
+
+            form.UICommands.RepoChangedNotifier.Notify();
+        }
+
+        private void MoreButton_Click(object sender, EventArgs e)
+        {
+            if (!(FindForm() is GitModuleForm form))
+            {
+                return;
+            }
+
+            switch (_state)
+            {
+                case State.Bisect:
+                    BisectEvent?.Invoke(this, EventArgs.Empty);
+                    break;
+                case State.Rebase:
+                    form.UICommands.StartTheContinueRebaseDialog(form);
+                    break;
+                case State.Patch:
+                    form.UICommands.StartApplyPatchDialog(form);
+                    break;
+            }
+        }
+
+        internal TestAccessor GetTestAccessor()
+            => new TestAccessor(this);
+
+        internal readonly struct TestAccessor
+        {
+            private readonly InteractiveGitActionControl _interactiveGitActionControl;
+
+            internal TestAccessor(InteractiveGitActionControl interactiveGitActionControl)
+            {
+                _interactiveGitActionControl = interactiveGitActionControl;
+            }
+
+            internal State State => _interactiveGitActionControl._state;
+            internal bool HasConflicts => _interactiveGitActionControl._hasConflicts;
+
+            internal void SetState(State state, bool conflicts)
+            {
+                var that = this;
+                ThreadHelper.JoinableTaskFactory.Run(async () =>
+                {
+                    await that._interactiveGitActionControl.SetStateAsync(state, conflicts);
+                });
+            }
```

Comment:
The `TestAccessor` is a proxy for private members, any smarts that may be necessary should be done in tests.
So we would have it:
```cs
internal void SetState(State state, bool conflicts) => _interactiveGitActionControl.SetState(state, conflicts);
```